auth_provider/tests.py

```diff
@@ -63,3 +69,36 @@ def test_login_page_require_admin_approval(self):
         res = self.client.post(reverse_lazy('auth_provider:login'), data=login_data)
         self.assertEqual(res.status_code, 200)
         self.assertContains(res, 'error')
+
+
+class RemoteProfileViewTests(TestCase):
+    def setUp(self) -> None:
+        self.client = Client()
+        get_user_model().objects.create_user(username='bob', password='password')
+
+        mock_json_response = json.loads(SAMPLE_REMOTE_AUTHOR)
+        mock_response = Response()
+        mock_response.json = MagicMock(return_value=mock_json_response)
+
+        mock_server = Server(
+            service_address="http://localhost:5555/api/v2",
+            username="hello",
+            password="no",
+        )
+        mock_server.get = MagicMock(return_value=mock_response)
+
+        with patch('servers.models.Server.objects') as MockServerObjects:
+            MockServerObjects.all.return_value = [mock_server]
+
+            self.client.login(username='bob', password='password')
+            res = self.client.get(
+                reverse(
+                    'accounts:remote_profile',
+                    kwargs={
+                        'url': 'http://localhost:5555/api/v2/authors/1/'
+                    }
+                )
+            )
+            self.assertEqual(res.status_code, 200)
+            self.assertContains(res, mock_json_response['display_name'])
+            self.assertContains(res, mock_json_response['github'])
```

Comment:
This seems like the test case logic, but it is inside of the `setup` function. Thinking that this should be moved to a `test_remote_profile_page` function so that it is a proper test case. Ex: 

```python
def test_remote_profile_page(self):
      self.client.login(username='bob', password='password')
      res = self.client.get(
            reverse(
                  'accounts:remote_profile',
                  kwargs={
                        'url': 'http://localhost:5555/api/v2/authors/1/'
                  }
            )
      )
      self.assertEqual(res.status_code, 200)
      self.assertContains(res, mock_json_response['display_name'])
      self.assertContains(res, mock_json_response['github'])
```
Not sure how that will affect the `patch` logic though.

Reply:
Patch only applies to the body so yes this should be moved only to the test case, we shouldn't assert anything other than for verification in `setUp`